sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,453 @@ def reduce_inequalities(inequalities, symbols=[]):
+    # over the y variables during Bland's rule
+    r_orig = [(False, j) for j in range(M.cols - 1)] # what Ferguson's introduction calls 'x' variables
+    s_orig = [(True, i)  for i in range(M.rows - 1)] # what Ferguson's introduction calls 'y' variables
+
+    R = r_orig.copy()
+    S = s_orig.copy()
+
+    # Phase 1: find a feasible solution or determine none exist
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            # We have found a feasible solution
+            break
+
+        # Find k, first row with a negative rightmost entry
+        for k in range(B.rows):
+            if B[k] < 0:
+                break
+
+        # Choose pivot column, j0
+        piv_cols = [j for j in range(A.cols) if A[k, j] < 0]
+        if not piv_cols:
+            raise InfeasibleLinearProgrammingError('The constraint set is empty!')
+        j0 = sorted(piv_cols, key=lambda c: R[c])[0] # Bland's rule
+
+        # Choose pivot row, i0
+        piv_rows = [i for i in range(A.rows) if A[i, j0] > 0 and B[i] > 0]
+        piv_rows.append(k)
+        i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    # Phase 2: starting at a feasible solution, pivot until we reach optimal solution
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+
+        # Choose a pivot column, j0
+        piv_cols = []
+        piv_cols = [j for j in range(C.cols) if C[j] < 0]
+        if not piv_cols:
+            break
+        j0 = sorted(piv_cols, key=lambda c: R[c])[0] # Bland's rule
+
+        # Choose a pivot row, i0
+        piv_rows = [i for i in range(A.rows) if A[i, j0] > 0]
+        if not piv_rows:
+            raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large values!')
+        i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    argmax = [None]*(M.cols-1)
+    argmin_dual = [None]*(M.rows-1)
+
+    for i, var in enumerate(R):
+        v = var[0]
+        n = var[1]
+        if v == False:
+            argmax[n] = 0
+        else:
+            argmin_dual[n] = M[-1, i]
+
+    for i, var in enumerate(S):
+        v = var[0]
+        n = var[1]
+        if v == True:
+            argmin_dual[n] = 0
+        else:
+            argmax[n] = M[i, -1]
+
+    return M[-1, -1], argmax, argmin_dual
+
+
+def linear_optimization_from_matrices(f, A, b):
+    """
+    Because of the duality of linear programming, there are two valid ways to
+    interpret this function. The first is as a solver for the standard
+    maximization problem:
+        Maximize Cx constrained to Ax <= B and x >= 0.
+    The second as is a solver for the standard minimization problem:
+        Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
```

Comment:
```suggestion
        Minimizing y^{T}B constrained to y^{T}A >= C and y >= 0.
```
But I think it would be better to stick to the conventions of the reference/Ferguson and let C be the transpose of what you have now.